### ISSUE TITLE
Accessibility: Change address button on cart page cannot be triggered with space bar

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43620]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The "Change Address" link on the cart page has been correctly marked as a button with ARIA; however, it cannot be triggered with the space bar.

Relevant code:
https://github.com/woocommerce/woocommerce/blob/758df4854d7b627fc26eb5c0860dc492dea6a06e/plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/shipping/calculator-button.tsx#L18C1-L18C1

### Expected behavior

Buttons should be able to be triggered with the space bar and Tab key. When turning a link into a button with ARIA, you need to add JavaScript keyboard handlers so that it can be triggered with the space bar.

[See the MDN docs for additional information and example code.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#accessibility_concerns)

### Actual behavior

Hitting the space bar on the Change Address button does not reveal the fields to change the address and instead scrolls users down the page.

### Steps to reproduce

On the cart page, focus on the "Change Address" button via the Tab key, then hit the space bar.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.
